inventory/internal/store/osupdaterun.go

```diff
@@ -0,0 +1,384 @@
+// SPDX-FileCopyrightText: (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+package store
+
+// osupdaterun.go  store information for OSUpdateRun objects
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+
+	"github.com/open-edge-platform/infra-core/inventory/v2/internal/ent"
+	our "github.com/open-edge-platform/infra-core/inventory/v2/internal/ent/osupdaterunresource"
+	compute_v1 "github.com/open-edge-platform/infra-core/inventory/v2/pkg/api/compute/v1"
+	inv_v1 "github.com/open-edge-platform/infra-core/inventory/v2/pkg/api/inventory/v1"
+	status_v1 "github.com/open-edge-platform/infra-core/inventory/v2/pkg/api/status/v1"
+	cl "github.com/open-edge-platform/infra-core/inventory/v2/pkg/client"
+	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/errors"
+	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/util"
+	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/util/collections"
+	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/validator"
+)
+
+var osUpRunResourceCreationValidators = []resourceValidator[*compute_v1.OSUpdateRunResource]{
+	protoValidator[*compute_v1.OSUpdateRunResource],
+	validateOSUpdateRunProto,
+	doNotAcceptResourceID[*compute_v1.OSUpdateRunResource],
+}
+
+func validateOSUpdateRunProto(in *compute_v1.OSUpdateRunResource) error {
+	if in.GetAppliedPolicy() == nil {
+		return errors.Errorfc(codes.InvalidArgument, "Applied Policy cannot be nil")
+	}
+	if in.GetInstance() == nil {
+		return errors.Errorfc(codes.InvalidArgument, "Instance cannot be nil")
+	}
+	if in.GetStatusIndicator() != status_v1.StatusIndication_STATUS_INDICATION_UNSPECIFIED {
+		return errors.Errorfc(codes.InvalidArgument, "OS Update Status Indicator should be unspecified")
+	}
+	if in.GetStatus() != "" {
+		return errors.Errorfc(codes.InvalidArgument, "OS Update Status should be nil")
+	}
+	if in.GetStatusDetails() != "" {
+		return errors.Errorfc(codes.InvalidArgument, "OS Update Status details should be nil")
+	}
+	if in.GetStatusTimestamp() != "" {
+		return errors.Errorfc(codes.InvalidArgument, "OS Update Status timestamp should be nil")
+	}
+	if in.GetStartTime() == "" {
+		return errors.Errorfc(codes.InvalidArgument, "OS Update Start timestamp should not be nil")
+	}
+	if in.GetEndTime() != "" {
+		return errors.Errorfc(codes.InvalidArgument, "OS Update End timestamp should be nil")
+	}
```

Comment:
Why all these validation? Most of these can be enforced via the protobuf validate library.
If you added this to enforce immutability, that is enforced at ent/DB layer, we don't need to write such validation.